push_action/run.py

```diff
@@ -188,17 +187,14 @@ def protected_branch(branch: str) -> str:
     """
     url = f"/repos/{os.getenv('GITHUB_REPOSITORY', '')}/branches/{branch}"
     response: "Dict[str, Any]" = api_request(url)  # type: ignore[assignment]
-    if response.get("protected", False):
-        if check_user_role("admin"):
```

Comment:
This should work. I only removed this line and this action worked with the Github App token.

Then I found out docker actions don't run on macos runners where I needed this action to do our iOS builds. So I had to rewrite this entire action in javascript 🙃. Used your stuff as a template so it was fairly smooth. Thanks!

Reply:
I have wanted to do this for a while; updating it to JS from Docker. If you want to contribute it back feel free, otherwise it's fair to release under a fork as a new action as well 👍